### PR TITLE
Recipe App

### PR DESCRIPTION
**Pull request by Jeddaa**
For the recipe app project we created:

- Login page and registration page with Devise.
- Food list that displays a list of food added by the logged-in user.
- Addition of new food by logged-in user.
- Recipes list that displays a list of recipes created by the logged-in user.
- Addition of new recipes by logged-in user.
- Delete recipe if owned by logged-in user.
- Public recipes list that displays a list of all public recipes ordered by newest.
- A toggle button that allows for a recipe to be made public or private.
- General shopping list view.
- Make sure there are no N+1 queries happening.
- A navigation menu that allows users to open all of the pages you created.
- Unit and integration tests

### CODE REVIEW

File: app/controllers/recipe_foods_controller.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  end
+
+  def show
+    @recipe_food = RecipeFood.find(params[:id])
+  end
+
+  def new
+    @recipe = Recipe.find(params[:recipe_id])
+    @recipe_food = RecipeFood.new
+  end
```
Comment (Zuhaib042) [inline app/controllers/recipe_foods_controller.rb:16]: - [ ] There is one big problem in your app. on the recipe details page, when I try to add an ingredient, I get to choose the ingredients only from the foods that I have chosen before. I think here the user is free to add new ingredients other than he added before. so thats one correction you need to make as per requriements.
 
Screenshot 2023-08-04 at 10 27 44 PM

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi @Jeddaa @coderacheal 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ Sign up and Log in with devise
✅ Good test coverage
✅ App's design is good
✅ Eliminating n+1 queries

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/foods/index.html.erb [hunk 2]
```diff
@@ -3,7 +3,7 @@
   <h1 style='color: rgb(48, 221, 48);'>Welcome back!</h1>
   <h3>This is what you currently have in your pantry</h3>
   <div class='add-ingredient-button-div'>
-    <%= button_to 'Add ingredient', new_food_path, class: 'btn', method: :get %>
+    <%= button_to 'Add Food', new_food_path, class: 'btn', method: :get %>
```
Comment (tufoinnkuo10) [inline app/views/foods/index.html.erb:6]: - [optional] Great job renaming the button. Please kindly add the size a little and the button text bolder so it's more visible and follows the design.

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @TEAM

Your project is complete! Great job for implementing the changes requested🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request✔️
- Devise works well✔️
- well tested✔️
- wireframe followed✔️
- No linter errors✔️✔️

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**